GitCommands/Git/GitModule.cs

```diff
@@ -161,6 +161,21 @@ bool HasGitModulesFile(string path)
         [CanBeNull]
         public GitModule SuperprojectModule { get; }
 
+        /// <summary>
+        /// If this module is a submodule, returns the top-most parent module, otherwise it returns itself.
+        /// </summary>
+        [NotNull]
+        public GitModule GetTopModule()
```

Comment:
```suggestion
        public IGitModule GetTopModule()
```

Reply:
As I wrote in a previous comment, changing GitModule to IGitModule induces too many changes for this PR. In a future change, this function should be added to IGitModule and return IGitModule, as you suggest. For now, it's a GitModule-specific function returning GitModule.

Reply:
Sorry, I must have missed it completely. Worth putting a note in the code so we know to address it

Reply:
I'll add a few TODOs for those functions in GitModule that need moving up to IGitModule.